MapboxCoreNavigation/OfflineRegionStatus.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+
+/**
+ * State transitions of an offline region.
+ *
+ *                  download()
+ *                      │
+ *                      │
+ *                   ┌──▼──────┐
+ *              ┌────▶ Pending ├─────────────delete()────┐
+ *              │    └───────▲─┘                         │
+ *              │      │  │  │                           │
+ *          download() │  │  └───download()──┐           │
+ *              │      │  │                  │           │
+ *              │      │  │           ┌──────┴─────┐     │
+ *              │      │  └─cancel()──▶ Incomplete ├──┐  │
+ *              │      │              └──────▲─────┘  │  │
+ *      ┌───────┴─┐  (scheduled)             │        │  │
+ *      │ Errored │    │                     │        │  │
+ *      └─▲──┬──▲─┘    │    ┌────cancel()────┘        │  │
+ *        │  │  │      │    │                   delete() │
+ *        │  │  │  ┌───▼─────────┐                    │  │
+ *        │  │  └──│ Downloading │───delete()──────┐  │  │
+ *        │  │     └─────────────┘                 │  │  │
+ *        │  │            │                        │  │  │
+ *        │  │       (completes)                 ┌ ▼ ─▼─ ▼ ┐
+ *        │  └────────────│───────────delete()───▶ Deleted
+ *        │               │                      └ ─▲─ ─▲─ ┘
+ *        │         ┌─────▼─────┐                   │   │
+ *     (failed)─────│ Verifying │─────delete()──────┘   │
+ *                  └───────────┘                       │
+ *                        │                         (succeeds)
+ *                   (successful)                       │
+ *                        │                             │
+ *                  ┌─────▼─────┐                  ┌────┴─────┐
+ *                  │ Available ├───────delete()───▶ Deleting │
+ *                  └─────▲─────┘                  └────┬─────┘
+ *                        │                             │
+ *                        └─────────download()──────────┘
```

Comment:
Can you check if this ASCII art appears correct in jazzy docs? I think we might need to wrap it in some markup to keep it as preformatted text.